R/estimateDiversity.R

```diff
@@ -149,23 +158,30 @@ NULL
 
 #' @rdname estimateDiversity
 #' @export
-setGeneric("estimateDiversity",signature = c("x"),
+setGeneric("estimateDiversity",signature = c("x", "tree"),
            function(x, abund_values = "counts",
                     index = c("shannon","gini_simpson", "inverse_simpson",
-                              "coverage", "fisher"),
-                    name = index, ...)
+                              "coverage", "fisher", "pd"),
```

Comment:
The default setting in the generic will overwrite the default in setMethod AFAIK.

So I think, it would be good to keep this as a separate function with its own generic for now.